Add TypeScript type definitions

theraccoonbear opened the pull request:
This is my first time writing type defs for an existing JS module.  In my testing this provides auto-completion in VSCode and makes the TS compiler happy when importing the module.

infusion commented:
Looks pretty complete now. I don't know if you want to support the `constructor (numerator: number, denominator: number);` syntax for all operator methods like add, sub, ... as well. The second thing is, you had the module name `Fraction` before. I like this better than with the .js suffix in the ts world. Does this cause any problems?

theraccoonbear commented:
Good call.  I've updated the methods to support the same arguments as the constructor.  I also renamed the module to `Fraction`.  Again, I am new to this, but I don't think the module name in this type def is strictly required.

infusion commented:
That looks pretty!

theraccoonbear commented:
@infusion I think this is complete at this point.  Please merge whenever you have a chance.  Thanks!
